### DISCUSSION TITLE
How to deploy VelaUX?

### DISCUSSION THREAD

**Discussion by dudicoco** [kubevela/kubevela#4366] [General] [answered]
The instructions for deploying VelaUX only include an option to do it via the CLI: https://kubevela.io/docs/reference/addons/velaux.

I would like to deploy it using standard k8s manifests or via a helm chart.
Is that possible?

**Comment by wonderflow** [comment]
There's an interesting way for your need https://kubevela.io/docs/platform-engineers/system-operation/vela-cli-image#integrate-vela-cli-image-with-kubernetes

**Comment by dudicoco** [comment, marked as answer]
Thanks @wonderflow.

It seems to me that kubevela needs to be modernized with declarative configuration.
Using api/cli to config a tool these days instead of a declarative config file feels very outdated.

I don't feel comfortable with installing a tool which requires running a k8s job to configure itself.

**Reply by wonderflow** [reply]
So, what are your real use case? I don't understand what's wrong with k8s job?

**Reply by wonderflow** [reply]
kubectl and helm are all cli tools. While k8s job is also the declarative way. I'm sorry that I really can't understand what is morden in your mind. It's interesting.

**Reply by dudicoco** [reply]
So, i'm not sure what the CLI addon actually does - does it configure the kubevela controller deployment to expose the UI on a certain port? Does it create a new deployment using the velaux docker image?

If it's the first option, I would expect the kubevela controller to either accept a flag/environment variable/config file to enable the UI.
If it's the second, I would like to have a k8s manifest of the velaux deployment object. According to https://github.com/kubevela/velaux/blob/main/Dockerfile and https://kubevela.io/docs/reference/addons/velaux#specify-the-addon-image it seems that it does run as its own separate container image.

**Reply by wonderflow** [reply]
Addon is actually a bunch of yaml, it will help you deploy. What if we have a command that help render these yamls:

```
vela addon enable velaux --dry-run
```

And it prints all those yamls, then you can apply them?

```
vela addon enable velaux --dry-run > addon-manifest.yaml
kubectl apply -f addon-manifest.yaml
```

**Reply by dudicoco** [reply]
Thanks @wonderflow, since i'm not interested in using the CLI could you please share the manifest here?
It would also help others who might be interested in declarative deployment.

**Reply by dudicoco** [reply]
@wonderflow i've tried using the vela cli command but it fails:
```sh
Error: unknown flag: --dry-run
```

**Reply by wonderflow** [reply]
@dudicoco Hi, you can refer to v1.6.0-beta.1 to use this command.

**Comment by geowalrus4gh** [comment]
We are not able to download UX via cli.

> Error: download index file from https://addons.kubevela.net failure Get "https://addons.kubevela.net/index.yaml": dial tcp connect: connection refused

Corporate firewall is not allowing to install velaUX. 
It would be great if you can provide the helm charts of velaUX or add it along with core.

**Reply by wonderflow** [reply]
Notice: the "hub.kubevela.net", "charts.kubevela.net" and "addons.kubevela.net" will not be served in the future.

For docker images in "hub.kubevela.net", replace it with "docker.io" or "ghcr.io/kubevela".
For helm charts in "charts.kubevela.net", replace it with "kubevela.github.io/charts".
For addons in "addons.kubevela.net", replace it with "kubevela.github.io/catalog/official".
You can run the following command to update your addon registry.

```
vela addon registry delete KubeVela
vela addon registry add KubeVela --type helm --endpoint=https://kubevela.github.io/catalog/official
vela addon registry add experimental --type helm --endpoint=https://kubevela.github.io/catalog/experimental
```

**Comment by kivra-joscan** [comment]
jumping back to this thread 3 years later, and im still thinks this should be included with the vela-core helm chart. Since the velaux is UX for kubevela that is deployed in the vela namespace it should be an optional resource to install in my opinion 

@wonderflow is that something that can be added or have been discussed before?
